ResearchKit/Common/ORKGraphView.m

```diff
@@ -0,0 +1,848 @@
+    self.xAxisPoints = [NSMutableArray new];
+    self.yAxisPoints = [NSMutableArray new];
+    self.xAxisTitles = [NSMutableArray new];
+    self.referenceLines = [NSMutableArray new];
+    self.pathLines = [NSMutableArray new];
+    self.dots = [NSMutableArray new];
+    self.tintColor = [UIColor colorWithRed:244/255.f green:190/255.f blue:74/255.f alpha:1.f];
+    self.shouldAnimate = YES;
+    self.hasDataPoint = NO;
+    self.panGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(handlePanGesture:)];
+    self.panGestureRecognizer.delaysTouchesBegan = YES;
+    self.panGestureRecognizer.delegate = self;
+    [self addGestureRecognizer:self.panGestureRecognizer];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(animateLayersSequentially)
+                                                 name:ORKGraphViewTriggerAnimationsNotification
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(refreshGraph)
+                                                 name:ORKGraphViewRefreshNotification
+                                               object:nil];
+    
+    [self setupViews];
+}
+
+- (void)setupViews {
+    self.plotsView = [UIView new];
+    self.plotsView.backgroundColor = [UIColor clearColor];
+    [self addSubview:self.plotsView];
+    
+    self.scrubberLine = [UIView new];
+    self.scrubberLine.backgroundColor = self.scrubberLineColor;
+    self.scrubberLine.alpha = 0;
+    [self addSubview:self.scrubberLine];
+    
+    self.scrubberLabel = [UILabel new];
+    self.scrubberLabel.font = [UIFont fontWithName:@"Helvetica-Light" size:12.0f];
+    self.scrubberLabel.alpha = 0;
+    self.scrubberLabel.layer.cornerRadius = 2.0f;
+    self.scrubberLabel.layer.borderColor = [UIColor darkGrayColor].CGColor;
+    self.scrubberLabel.layer.borderWidth = 1.0f;
+    self.scrubberLabel.textAlignment = NSTextAlignmentCenter;
+    self.scrubberLabel.frame = CGRectMake(2, 0, 100, 20);
+    self.scrubberLabel.backgroundColor = [UIColor colorWithWhite:0.98 alpha:0.8];
+    [self addSubview:self.scrubberLabel];
+    
+    self.scrubberThumbView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, [self scrubberThumbSize].width, [self scrubberThumbSize].height)];
+    self.scrubberThumbView.layer.borderWidth = 1.0;
+    self.scrubberThumbView.backgroundColor = self.scrubberThumbColor;
+    self.scrubberThumbView.layer.borderColor = [UIColor darkGrayColor].CGColor;
+    self.scrubberThumbView.alpha = 0;
+    [self addSubview:self.scrubberThumbView];
+}
+
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:ORKGraphViewTriggerAnimationsNotification object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:ORKGraphViewRefreshNotification object:nil];
```

Comment:
I guess just use `[[NSNotificationCenter defaultCenter] removeObserver:self]` would be able avoid tracking the notification names in the future.